src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/AlertAddressForm.tsx

```diff
@@ -46,31 +81,174 @@ export const AlertAddressForm = (props: Props) => {
       });
     },
   });
+
+  const handleMonthlyMonitorReportToggle = () => {
+    const newValue = !activateMonthlyMonitorReport;
+    setActivateMonthlyMonitorReport(newValue);
+    void fetch("/api/v1/user/update-comm-option", {
+      method: "POST",
+      body: JSON.stringify({
+        instantBreachAlerts: commsValue(),
+        monthlyMonitorReport: newValue,
+      }),
+    }).then(() => {
+      void session.update();
```

Comment:
This line surprised me - could you add a comment describing why it is necessary?

Reply:
Ah, I see there was such a comment in the collapsed code above already:

```suggestion
    }).then(() => {
      // Fetch a new token with up-to-date subscriber info - specifically,
      // with this setting updated.
      void session.update();
```